testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -324,7 +324,11 @@ def pillar_get(key, minion)
 
 Then(/^the pillar data for "([^"]*)" should be empty on "([^"]*)"$/) do |key, minion|
   output, _code = pillar_get(key, minion)
-  raise "Output has more than one line: #{output}" unless output.split("\n").length == 1
+  repeat_until_timeout(timeout: DEFAULT_TIMEOUT, message: "Output has more than one line: #{output}", report_result: true) do
+    break unless output.split("\n").length != 1
+    sleep 1
+    output, _code = pillar_get(key, minion)
+  end
```

Comment:
Only one suggestion.
Maybe could be better to have:

```
Then(/^the pillar data for "([^"]*)" should be empty on "([^"]*)"$/) do |key, minion|
  repeat_until_timeout(timeout: DEFAULT_TIMEOUT, message: "Output has more than one line: #{output}", report_result: true) do
    output, _code = pillar_get(key, minion)
    break unless output.split('\n').length != 1
    sleep 1
  end
end
```

Reply:
Rubocop does not complain but the code does complain when being run as `output` is not yet known